internal/juju/applications.go

```diff
@@ -435,6 +541,28 @@ func (c applicationsClient) UpdateApplication(input *UpdateApplicationInput) err
 		}
 	}
 
+	// process expose
+	if input.Unexpose {
+		// TODO: (2022-08-10) right now we only unexpose all the possible endpoints.
+		// Additional logic must be set up to specify what endpoints
+		// should be unexpose in case this is simply a change, not a
+		// complete removal.
+		if err := applicationAPIClient.Unexpose(input.AppName, nil); err != nil {
```

Comment:
Can't we use the second argument of the `client.Unexpose` to list the specific endpoints? (i.e. for the `--endpoints` flag).

Reply:
Not now, because this information is populated by the Terraform provider during updates and for that, we need to compute the delta between the previous and the current plan version. I leave that logic for next PRs.